2019/d01p1FuelCounterUpper.d

```diff
@@ -0,0 +1,7 @@
+unittest {
+  import std.algorithm;
+
+  auto input = [12, 14, 1969, 100756];
+  auto fuelTotal = input.map!"a / 3 - 2".sum;
```

Comment:
This line is where D shows a hint of its magic, specifically Uniform Function Calling Syntax (pronounced "dot") and that you can safely mix strings of code with code.

`map` is a template function that applies a function over a range of values. Here the mass-to-fuel function is provided as a string. This is not evaluated at runtime - the compiler is compiling this, with all the safeguards thereof. `a` is the name of the function's argument representing the input value.

`sum` is a function which takes an input range and returns the sum of all its values.

So, the total amount of fuel is the sum of the mass-to-fuel function applied to each of the input module mass values.